lib/util/http/index.ts

```diff
@@ -126,20 +127,34 @@ export class Http<GetOptions = HttpOptions, PostOptions = HttpPostOptions> {
         return resolveResponse<T>(cachedRes, options);
       }
     }
-    const startTime = Date.now();
-    const promisedRes = got<T>(url, options);
+    let startTime: number;
+    const queueTask = (): Promise<Response<T>> => {
+      const queueTime = Date.now();
+      const queueDuration = queueTime - startTime;
+      return got<T>(url, options).then((resp) => {
+        const respTime = Date.now();
+        const duration = respTime - queueTime;
+        const httpRequests = memCache.get('http-requests') || [];
+        httpRequests.push({
+          method: options.method,
+          url,
+          duration,
+          queueDuration,
+        });
+        memCache.set('http-requests', httpRequests);
+        return resp;
+      });
+    };
+
+    const queue = getQueue(url);
+    startTime = Date.now();
```

Comment:
Is there the possibility that the startTime value could be overwritten by concurrent requests?

Reply:
I don't think so, because we create `startTime` closure over corresponding `queueTask()` in each request.

Reply:
Though, I made this code simpler